src/modules/Users.pm

```diff
@@ -5444,8 +5444,8 @@ sub CheckUser {
 
     if ($error eq "") {
 	# do not check pw when it wasn't changed - must be tested directly
-	if (defined ($user{"userPassword"}) ||
-	    ($user{"what"} || "") eq "add_user") {
+	if ((defined ($user{"userPassword"}) ||
+	    ($user{"what"} || "") eq "add_user") && (!$user{"encrypted"})) {
```

Comment:
The condition was already quite a nightmare. But now it's even worse. I'd appreciate a comment or using an intermediate boolean variable or something. I can't imagine myself debugging this in 2 years.